cmake/toolchains/GCC_ARM.cmake

```diff
@@ -15,6 +15,7 @@ list(APPEND link_options
     "-lgcc"
     "-lnosys"
     "-Wl,--end-group"
+    "-specs=nosys.specs"
```

Comment:
Why was this added?

Reply:
It should be described in the commit message. lib nosys but that is not sufficient as there are specs as well and did not reflect the library selection: librdimon.a or libnosys.a.

If we remove this, we get linker complaining about not finding _kill and other functions.

There are 3 specs available rdimon, nosys or nano by GCC ARM.

Reply:
Sorry I did not look at individual commits.

Was the same option added by Mbed CLI? If not, are we masking something else by including this flag?

Reply:
Some exporters had it but I could not find it in tools. Please fetch this PR and compare flags between this and master, to be certqin

Reply:
I am investigating the cause of the failure because all three profiles were working fine prior to your changes.

Reply:
Thanks !